### PR TITLE
fix: harden share validation, eliminate FooterBar O(n) scan, add lane drag-to-reorder

### PR DESCRIPTION
**Pull request by xaedyn**
### CodeRabbit fixes (last 2 findings from PRs #1, #4, #5)

- **Share validation hardening** — `validateSharePayload` now rejects non-HTTP(S) URLs (`javascript:`, `data:`, relative paths), non-finite/negative numeric fields, oversized arrays (endpoints > 50, results > 50, samples > 10k), and invalid `corsMode` values
- **FooterBar O(n) elimination** — error/timeout counts tracked incrementally in `measurementStore` via `addSample`/`addSamples`/`loadSnapshot`/`reset`; FooterBar reads pre-computed values (1 line replaces 11-line O(n) derived scan)

### Lane drag-to-reorder (new feature)

- **Grip handle** — 6-dot SVG grip on left edge of stats panel (full mode) and compact header. Hidden when only 1 endpoint.
- **Pointer event drag** — `pointerdown` on grip captures pointer, `pointermove` translates dragged card via `--drag-translate` CSS custom property, neighbors shift with spring easing
- **Drop commits reorder** — `pointerup` transitions card to target slot, waits for spring settle, then commits via `endpointStore.reorderEndpoint(from, to)`. Order persists via existing localStorage save-on-change.
- **Apple-level animation polish** — pickup overshoot keyframe (scale 1.0→1.03→1.01), spring easing on neighbor shifts (`cubic-bezier(0.34, 1.56, 0.64, 1)`), settling de-lift transition on drop, transition suppression during DOM reorder to prevent neighbor twitch. Zero dependencies.

## Test plan

- [x] 17 validation edge-case tests (URL schemes, Infinity/NaN/negative, array bounds, corsMode)
- [x] 11 store counter tests (addSample, addSamples, reset, loadSnapshot)
- [x] 8 reorder store tests (forward, backward, multi-hop, out-of-bounds, data preservation)
- [x] 5 grip handle tests (render, aria-label, compact mode, data-endpoint-id, showGrip=false)
- [x] 4 LanesView tests (container, lanes render, grip handles, single-endpoint hidden)
- [x] Visually tested drag-to-reorder in Chrome — pickup, drag, drop settle, persistence
- [x] Full suite: 371 tests passing
- [x] Typecheck clean, lint clean

## Summary by CodeRabbit

* **New Features**
  * Error and timeout counters are tracked and displayed in results/footer.
  * Drag-to-reorder lanes with a visible grip handle for reordering endpoints.

* **Improvements**
  * Hardened validation for shared measurement payloads (URL formats, numeric settings, corsMode, and size limits).
  * Snapshot loading now ensures counters are consistent with displayed results.

* **Tests**
  * Expanded unit tests for payload validation, counters, and reorder behavior.

### CODE REVIEW

File: src/lib/share/share-manager.ts [hunk 1]
```diff
@@ -29,41 +29,55 @@ function isFiniteNumber(v: unknown): boolean {
...
   const settings = obj['settings'];
   if (settings === null || typeof settings !== 'object') return null;
   const s = settings as Record<string, unknown>;
-  if (!isFiniteNumber(s['timeout']) || !isFiniteNumber(s['delay']) || !isFiniteNumber(s['cap'])) {
-    return null;
-  }
+  if (!isNonNegativeFiniteNumber(s['timeout'])) return null;
+  if (!isNonNegativeFiniteNumber(s['delay'])) return null;
+  if (!isNonNegativeFiniteNumber(s['cap'])) return null;
+  if (s['corsMode'] !== 'no-cors' && s['corsMode'] !== 'cors') return null;
```
[the remark this answers is not in the document]
Comment (xaedyn) [reply src/lib/share/share-manager.ts:66]: Fixed in b2affac. Added `isNonNegativeFiniteNumber` checks for optional `burstRounds` and `monitorDelay`.

File: src/lib/share/share-manager.ts [hunk 2]
```diff
@@ -29,41 +29,55 @@ function isFiniteNumber(v: unknown): boolean {
...
       for (const sample of r['samples'] as unknown[]) {
         if (sample === null || typeof sample !== 'object') return null;
-        const s = sample as Record<string, unknown>;
-        if (typeof s['round'] !== 'number' || typeof s['latency'] !== 'number' || typeof s['status'] !== 'string') return null;
+        const samp = sample as Record<string, unknown>;
+        if (
+          typeof samp['round'] !== 'number' ||
+          typeof samp['latency'] !== 'number' ||
+          typeof samp['status'] !== 'string'
+        ) return null;
```
[the remark this answers is not in the document]
Comment (xaedyn) [reply src/lib/share/share-manager.ts:82]: Fixed in b2affac. Sample validation now uses `isNonNegativeFiniteNumber` for round/latency and checks status against the `SampleStatus` union (`'ok' | 'timeout' | 'error'`).

File: src/lib/stores/measurements.ts [hunk 3]
```diff
@@ -20,8 +20,29 @@ const INITIAL_STATE: MeasurementState = {
...
+  let errorCount = 0;
+  let timeoutCount = 0;
+  for (const ep of Object.values(endpoints)) {
+    for (const sample of ep.samples) {
+      if (sample.status === 'error') errorCount++;
+      if (sample.status === 'timeout') timeoutCount++;
+    }
+  }
+  return { errorCount, timeoutCount };
+}
```
[the remark this answers is not in the document]
Comment (xaedyn) [reply src/lib/stores/measurements.ts:44]: Fixed in b2affac. `removeEndpoint` now calls `recomputeCounts(rest)` after removing the endpoint.

File: src/lib/components/Lane.svelte [hunk 4]
```diff
@@ -60,6 +77,21 @@
...
 >
   <div class="lane-panel" class:sr-only={compact}>
+    {#if showGrip}
+      <button
+        class="lane-grip"
+        aria-label="Reorder lane"
+        data-endpoint-id={endpointId}
+        type="button"
+        onpointerdown={onGripPointerDown}
+      >
```
[the remark this answers is not in the document]
Comment (xaedyn) [reply src/lib/components/Lane.svelte:96]: Fixed in b2affac. Compact mode now renders a single interactive grip in the compact header (no `aria-hidden` ancestor). The sr-only panel skips the grip button when `compact` is true.

File: src/lib/components/LanesView.svelte [hunk 5]
```diff
@@ -108,6 +108,95 @@
...
+
+    const maxUp = -(dragState.fromIndex * dragState.cardHeight);
+    const maxDown = (count - 1 - dragState.fromIndex) * dragState.cardHeight;
+    const clampedDelta = Math.max(maxUp, Math.min(maxDown, deltaY));
+
+    const sign = clampedDelta >= 0 ? 1 : -1;
+    const newToIndex = Math.max(
+      0,
+      Math.min(count - 1, dragState.fromIndex + Math.round(clampedDelta / dragState.cardHeight)),
+    );
```
[the remark this answers is not in the document]
Comment (xaedyn) [reply src/lib/components/LanesView.svelte:197]: Fixed in b2affac. Grip is gated off in `compact-2col` mode: `showGrip={endpoints.length > 1 && layoutMode !== 'compact-2col'}`.

File: src/lib/stores/endpoints.ts [hunk 6]
```diff
@@ -64,6 +64,25 @@ function createEndpointStore() {
...
+        ) {
+          return endpoints;
+        }
+        const next = [...endpoints];
+        const moved = next.splice(fromIndex, 1)[0];
+        if (moved === undefined) return endpoints;
+        next.splice(toIndex, 0, moved);
+        return next;
+      });
+    },
```
[the remark this answers is not in the document]
Comment (xaedyn) [reply src/lib/stores/endpoints.ts]: Fixed in b2affac. Switched `reorderEndpoint` to ID-based API (`fromId: string, toId: string`). The store resolves indices internally from the full array, so filtered-list callers can't corrupt ordering.

File: src/lib/components/LanesView.svelte [hunk 7]
```diff
@@ -108,6 +108,138 @@
...
+
+    dragState = {
+      pointerId: e.pointerId,
+      fromIndex,
+      startY: e.clientY,
+      cardHeight,
+      toIndex: fromIndex,
+    };
+    dragOffsets = {};
+  }
```
[the remark this answers is not in the document]
Comment (xaedyn) [reply src/lib/components/LanesView.svelte:180]: Fixed in b2affac. Added `reorderPending` flag — set before the 280ms timeout, cleared after DOM settles. `handleGripPointerDown` returns early when the flag is set.

File: src/lib/components/Lane.svelte [hunk 8]
```diff
@@ -60,6 +83,21 @@
...
+        <svg width="10" height="14" viewBox="0 0 10 14" aria-hidden="true">
+          {#each GRIP_DOTS as [cx, cy], i (i)}
+            <circle {cx} {cy} r="1.5" fill="currentColor" />
+          {/each}
+        </svg>
+      </button>
```
[the remark this answers is not in the document]
Comment (xaedyn) [reply src/lib/components/Lane.svelte:102]: Fixed in 3ec43a7. Added `onGripKeyDown` handler — ArrowUp/ArrowDown triggers immediate reorder with focus preservation.

File: src/lib/components/LanesView.svelte [hunk 9]
```diff
@@ -108,6 +108,142 @@
...
+        dragOffsets = {};
+        endpointStore.reorderEndpoint(endpoints[fromIndex].id, endpoints[toIndex].id);
+        // Re-enable transitions after the DOM settles
+        requestAnimationFrame(() => {
+          suppressTransition = false;
+          reorderPending = false;
+        });
+      }, 280); // matches the 280ms settling transition duration
+    });
+  }
```
[the remark this answers is not in the document]
Comment (xaedyn) [reply src/lib/components/LanesView.svelte:273]: Fixed in 3ec43a7. Added `pointercancel` handler on the lanes container that clears all drag state.

File: src/lib/components/LanesView.svelte [hunk 10]
```diff
@@ -108,6 +108,142 @@
...
+
+    // Wait for the spring transition to finish, then commit the reorder
+    reorderPending = true;
+    requestAnimationFrame(() => {
+      setTimeout(() => {
+        // Suppress transitions so neighbors don't twitch during DOM reorder
+        suppressTransition = true;
+        settlingIndex = null;
+        dragOffsets = {};
+        endpointStore.reorderEndpoint(endpoints[fromIndex].id, endpoints[toIndex].id);
```
[the remark this answers is not in the document]
Comment (xaedyn) [reply src/lib/components/LanesView.svelte]: Fixed in 3ec43a7. `fromId`/`toId` are now captured before the settling animation, outside the delayed closure.

File: src/lib/share/share-manager.ts [hunk 11]
```diff
@@ -29,41 +29,57 @@ function isFiniteNumber(v: unknown): boolean {
...
   if (!Array.isArray(obj['endpoints'])) return null;
+  if ((obj['endpoints'] as unknown[]).length > 50) return null;
 
-  // Validate each endpoint has required fields
   for (const ep of obj['endpoints'] as unknown[]) {
     if (ep === null || typeof ep !== 'object') return null;
     const e = ep as Record<string, unknown>;
-    if (typeof e['url'] !== 'string') return null;
+    if (!isHttpUrl(e['url'])) return null;
     if ('enabled' in e && typeof e['enabled'] !== 'boolean') return null;
```
[the remark this answers is not in the document]
Comment (xaedyn) [reply src/lib/share/share-manager.ts]: Fixed in 3ec43a7. Changed to `typeof e['enabled'] !== 'boolean'` — missing `enabled` now correctly rejects the payload.
